usr/share/rear/conf/default.conf

```diff
@@ -1329,22 +1329,47 @@ PXE_CONFIG_URL=
 PXE_CONFIG_PREFIX=rear-
 #
 # TFPT IP server. needed to create grub menu when PXE_CONFIG_GRUB_STYLE=y
-# If not set, we gonna try to get TFTP IP from PXE_TFTP_URL
+# If not set, we gonna try to get TFTP IP from PXE_TFTP_UPLOAD_URL
 PXE_TFTP_IP=
 #
 # where should we put the TFTP files ? (legacy way)
 PXE_TFTP_PATH=/var/lib/rear/output
 # where should we put the TFTP files ? (URL style)
-PXE_TFTP_URL=
+#PXE_TFTP_URL: deprecated as of 2.8. Use PXE_TFTP_UPLOAD_URL
+PXE_TFTP_UPLOAD_URL=
 #
 # prefix for PXE files, e.g. the hostname
 PXE_TFTP_PREFIX=$HOSTNAME.
 #
+# Optional support for downloading kernel and initrd by HTTP, thus making
+# downloads considerably faster.
+# First, you have to use lpxelinux.0 instead of pxelinux.0, the former supports
+# HTTP downloads. It's compatible, just replace pxelinux.0 in your TFTP
+# directory or set bootfile-name/filename to lpxelinux.0 in your DHCP server.
+# PXE_CONFIG_GRUB_STYLE must NOT be set to yes.
+# PXE_CONFIG_URL must be set and point to your pxelinux.cfg directory using nfs or any other mountable scheme.
+# PXE_TFTP_UPLOAD_URL must still be set as it provides all files still being downloaded by TFTP.
+# PXE_HTTP_UPLOAD_URL must be set and point to your web server root directory using nfs or any other mountable scheme.
+# PXE_HTTP_DOWNLOAD_URL must be set and point to your web server root directory using http.
+# PXE_HTTP_DOWNLOAD_URL must point to the same directory as PXE_HTTP_UPLOAD_URL.
+# You may set OUTPUT_PREFIX_PXE, but keep in mind it will be added to
+# PXE_TFTP_UPLOAD_URL, PXE_HTTP_UPLOAD_URL and PXE_HTTP_DOWNLOAD_URL.
+# Example:
+#    OUTPUT=PXE
+#    PXE_CONFIG_GRUB_STYLE=
+#    OUTPUT_PREFIX_PXE=rear_kernels/$HOSTNAME
+#    PXE_CONFIG_URL=nfs://tftp-server/var/lib/tftpboot/pxelinux.cfg
+#    PXE_TFTP_UPLOAD_URL=nfs://tftp-server/var/lib/tftpboot
+#    PXE_HTTP_UPLOAD_URL=nfs://web-server/var/www/html
+#    PXE_HTTP_DOWNLOAD_URL=http://web-server
+PXE_HTTP_UPLOAD_URL=
```

Comment:
In default.conf description and example look contradicting
because the description reads
```
# PXE_HTTP_DOWNLOAD_URL must be set and point to your web server root directory using http.
# PXE_HTTP_DOWNLOAD_URL must point to the same directory as PXE_HTTP_UPLOAD_URL.
```
while the example shows
```
#    PXE_HTTP_UPLOAD_URL=nfs://web-server/var/www/html
#    PXE_HTTP_DOWNLOAD_URL=http://web-server
```
So "web server root directory" versus
"same directory as PXE_HTTP_UPLOAD_URL" (i.e. "var/www/html")
is contradicting or at least looks confusing.

Reply:
"web server root directory" is a term used by web server administrators. In httpd you configure it with `DocumentRoot` while in nginx you configure it with `root`.
You actually have to **upload** to a directory `/var/www/html`, the `DocumentRoot`.
While **downloading**, the web server automatically adds `DocumentRoot` in front of the path and serves from that directory.

I.e. for supplying a file `xyz` you have to write to `/var/www/html/xyz`. When you request `http://web-server/xyz` you will be served with `/var/www/html/xyz`.

The only ones not confused by this are web server administrators. :-)

I tried to clear it up a little by referring to `DocumentRoot` in the description of PXE_HTTP_UPLOAD_URL and explicitly stating that PXE_HTTP_UPLOAD_URL and PXE_HTTP_DOWNLOAD_URL must *logically* point to the same directory.

Reply:
@robertdahlem 
thank you for making it clear.
I suspected it is something like that (it was ages ago
when I had set up a web server only to try out something)
but now it is clear ('DocumentRoot' is the "magic word").